functions/Get-DbaDiagnosticInfo.ps1

```diff
@@ -0,0 +1,129 @@
+
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+.LINK
+https://dbatools.io/Get-DbaDiagnosticInfo
+
+.EXAMPLE
+Get-DbaDiagnosticInfo
+
+This will collect basic information about the computer where DBAtools is being executed from. It will then show the information in the conslole and at the same time COPY it. After running the cmdlet just paste it where ever you are reporting the issue: GitHub issues, email, or SQLcommunity slack channel etc.
+
+.EXAMPLE   
+Get-DbatoolsEnvironmentInfo -Detailed
+
+This will show MORE DETAILED information about the computer where DBAtools is being executed from.
+    
+#>
+    Param (
+        #[string]$ComputerName = "$ENV:COMPUTERNAME",
+        [switch]$Detailed,
+        [switch]$Clip
+    )
+    
+    try
+    {
+        
+        # #Write-Verbose "Collecting information about the computer where DBAtools is excuted from"
+
+        Write-Verbose "Getting local enivornment information"
+        $localinfo = @{ } | Select-Object OSVersion, OsArchitecture,PowerShellVersion, PowerShellArchitecture, DbaToolsVersion, ModuleBase, CLR, SMO, DomainUser, RunAsAdmin, isPowerShellISE
+        
+        $localinfo.OSversion = [environment]::OSVersion.Version.ToString() 
+        $OsVersion = (Get-CimInstance CIM_OperatingSystem).Caption.ToString()
+        $localinfo.OSversion = $OsVersion + "(" + $localinfo.OSversion + ")"
+        
+        $localinfo.OsArchitecture = (Get-CimInstance CIM_OperatingSystem).OSArchitecture.ToString()
+
+        $localinfo.PowerShellVersion = $PSVersionTable.PSversion.ToString()
+
+        #If([Environment]::Is64BitProcess -eq $True) #Works on Powershell 3.0/.Net 4 and above
+        #$env:PROCESSOR_ARCHITECTURE -eq 'AMD64' 
+        If([Intptr]::Size -eq 8) {
+            $localinfo.PowerShellArchitecture = "64-bit PowerShell"
+        }
+        else{
+            $localinfo.PowerShellArchitecture = "32-bit PowreShell"
+        }
+
+        $localinfo.DbaToolsVersion = (Get-Module dbatools).Version.ToString()
+        $localinfo.ModuleBase = (Get-Module dbatools).ModuleBase.ToString() -replace "\\users\\\w+\\","\users\...\"
+        $localinfo.CLR = $PSVersionTable.CLRVersion.ToString()
+     
+        $smo = (([AppDomain]::CurrentDomain.GetAssemblies() | Where-Object { $_.Fullname -like "Microsoft.SqlServer.SMO,*" }).FullName -Split ", ")[1]
+        $localinfo.SMO = $smo.TrimStart("Version=")
+        
+        $localinfo.DomainUser = $env:computername -ne $env:USERDOMAIN
+        $localinfo.RunAsAdmin = ([Security.Principal.WindowsPrincipal][Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole] "Administrator")
+        $localinfo.isPowerShellISE = [Environment]::CommandLine -match 'ise.exe'  
+                
```

Comment:
If .Net version is important info that dbatools team will use when troubleshooting bug reports, sure we can add it.